src/holodeck/environments.py

```diff
@@ -470,11 +470,22 @@ def tick(self, num_ticks=1):
             self._command_center.handle_buffer()
 
             self._client.release()
-            self._client.acquire()
+            self._acquire_catch_crash()
             state = self._default_state_fn()
 
         return state
 
+    def _acquire_catch_crash(self):
```

Comment:
This is great! Should be helpful with trying to pin down the semaphore busy error.